velox/docs/functions/spark/json.rst

```diff
@@ -24,14 +24,16 @@ JSON Functions
     Casts ``jsonString`` to an ARRAY, MAP, or ROW type, with the output type 
     determined by the expression. Returns NULL, if the input string is unparsable.
     Supported element types include BOOLEAN, TINYINT, SMALLINT, INTEGER, BIGINT, 
-    REAL, DOUBLE, VARCHAR, ARRAY, MAP and ROW. When casting to ARRAY or MAP, 
+    REAL, DOUBLE, DATE, VARCHAR, ARRAY, MAP and ROW. When casting to ARRAY or MAP, 
     the element type of the array or the value type of the map must be one of 
     these supported types, and for maps, the key type must be VARCHAR. Casting 
     to ROW supports only JSON objects. ::
         
         SELECT from_json('{"a": true}'); -- {'a'=true} // Output type: ROW({"a"}, {BOOLEAN()})
         SELECT from_json('{"a": 1}'); -- {'a'=1} // Output type: ROW({"a"}, {INTEGER()})
         SELECT from_json('{"a": 1.0}'); -- {'a'=1.0} // Output type: ROW({"a"}, {DOUBLE()})
+        SELECT from_json('{"a":"2021-7-1T"}''); -- {'a'="2021-07-01"} // Output type: ROW({"a"}, {DATE()})
```

Comment:
```
spark-sql (default)> SELECT from_json('{"a":"2021-7-1T"}'');

[PARSE_SYNTAX_ERROR] Syntax error at or near ''': extra input '''.(line 1, pos 36)

== SQL ==
SELECT from_json('{"a":"2021-7-1T"}'');
------------------------------------^^^
```
Changed to this but another exception, looks like we need to specify the schema for it.
```
spark-sql (default)> SELECT from_json('{"a":"2021-7-1T"}');
[WRONG_NUM_ARGS.WITHOUT_SUGGESTION] The `from_json` requires [2, 3] parameters but the actual number is 1. Please, refer to 'https://spark.apache.org/docs/latest/sql-ref-functions.html' for a 
```

```
spark-sql (default)> SELECT from_json('{"a":"2021-7-1T"}', 'a Date');
{"a":2021-07-01}
Time taken: 1.172 seconds, Fetched 1 row(s)
```

Reply:
This function has different signatures in Velox and Spark.

In Spark, the second parameter is the schema of the result type.
In Velox, the result type is inferred from the expression, so the second parameter is not needed.

Reply:
Can you add this explanation to document?

Reply:
Done, thanks.